src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -137,11 +137,11 @@ static bool allInterval(Interval interval, BinaryOperatorKind relation, T value,
                                     return true;
                                 case LessThanOrEqual when tc.Related(LessThan, value, minValue):
                                     return false;
-                                case LessThanOrEqual when tc.Related(LessThanOrEqual, maxValue, value):
+                                case LessThanOrEqual when tc.Related(LessThan, maxValue, value):
```

Comment:
I remember that you had some direct tests on value set and value set factory. Could you add tests at that level for the changes here (ie. to test more directly)?

Reply:
There are such tests included here, however all of this code is thrown away in [https://github.com/dotnet/roslyn/pull/42886](https://github.com/dotnet/roslyn/pull/42886)

---
In reply to: [400463170](https://github.com/dotnet/roslyn/pull/42313#discussion_r400463170) [](ancestors = 400463170)